### PR TITLE
add blog about ui-demo

### PR DESCRIPTION
**Pull request by dust629**
### What changes are included in the pull request?

* add blog article about our new UI Tool
* add one more feature in doc/feature.md$
* add metadata in part3 

### Why are the changes needed?
* as a news to announce the UI Tool for interested readers
* explain metadata concept

### CODE REVIEW

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+description: Data Pipeline Visualization Tool
+slug: sdl-uidemo
+authors:
+  - name: Shasha Jiang
+    url: https://github.com/dust629
+tags: [DataCatalog, DataVisualization, DataFlow, DataLineage, DataManagement, DataPipeline, TechRevolution, GitHub OpenSource]
+hide_table_of_contents: false
+---
+
+We are thrilled to announce the launch of our new [UI tool](https://ui-demo.smartdatalake.ch/) designed to revolutionize the way visualize, track, and manage data flow! This extension of SDLB  acts as a user-friendly data catalog that showcases dependencies through a DAG, along with metadata for data objects and actions. Our UI tool is a step forward in data pipeline management, providing enhanced visibility and control. Whether you are a data engineer, analyst, or scientist, this tool will enhance your workflow management experience, allowing you to optimize processes and achieve exceptional results.
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
We are thrilled to announce the launch of our new [UI tool](https://ui-demo.smartdatalake.ch/) designed to revolutionize the way we visualize, track, and manage data pipelines! This addition to SDLB acts as a user-friendly technical data catalog that showcases dependencies through a graph, along with metadata about data pipeline execution (workflows). Our UI tool is a step forward in data pipeline management, providing enhanced visibility and control. Whether you are a data engineer, analyst, or scientist, this tool will enhance your abilities to management data pipelines, allowing you to be more productive and achieve exceptional results.

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+<!--truncate-->
+
+## 🌟 Explore Our UI Tool in Action
+
+Now Check out our [UI-Demo](https://ui-demo.smartdatalake.ch/), where the data pipeline in the [getting-started](../../docs/getting-started/setup) section with the part 3 [config file](../../docs/getting-started/config-examples/application-part3-with-meatdata.conf) is used as an example for our UI tool. Witness firsthand how our tool simplifies data pipeline management and enhances your workflow efficiency.
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
Now Check out our [UI-Demo](https://ui-demo.smartdatalake.ch/), where the data pipeline from the [getting-started](../../docs/getting-started/setup) section part 3 [config file](../../docs/getting-started/config-examples/application-part3-with-meatdata.conf) is used as show case. 
```
In my opinion the last sentence "Witness firsthand how our tool simplifies data pipeline management and enhances your workflow efficiency." is too much / repetitive...

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+![home](home.png "UI Homepage")
+<!--truncate-->
+
+Thanks to its user-friendly interface and powerful features, managing complex workflows has never been easier.
+
+## 📊 Comprehensive Workflow Dashboard
+Get a bird's eye view of your entire workflow! Our intuitive dashboard visualizes intricate details for each run, including execution time, duration, and the feed used. Dive deep into the timeline of  involved actions, with the ability to access detailed data by simply clicking on each action.
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
Get a bird's eye view of your entire workflow! Our intuitive dashboard visualizes intricate details for each run, including execution time, duration, and the feed configuration used. Deep dive into the timeline of involved actions, with the ability to access detailed data by simply clicking on each action.
```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+![Timeline of Action](timeline.png "Timeline of Action: Description or Caption")
+
+
+## 🔍 Config Viewer with Rich Content
+Explore an expansive Config Viewer that contains all the information from your config file. And extra information you provide as documentation. You can find:
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
Explore an expansive Configuration Viewer that contains all the information from your config file, and extra information you provide as documentation. You can find:

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+* ###  Markdown Descriptions:
+  Our tool supports markdown files as descriptions, enabling you to add detailed information about actions or objects. These descriptions can include images, hyperlinks, and other formatting options, serve as excellent documentation, aiding your understanding of the workflow's structure and function. Moreover, you can utilize descriptions to add labels by incorporating metadata in the config file, enhancing organization and clarity.
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
  SDLB supports creating descriptions as markdown files, enabling you to provide detailed information about actions or objects. These descriptions can include images, hyperlinks, and other formatting options. They serve as excellent documentation that is versionned together with your code, aiding your understanding of the workflow's structure and function. Moreover, you can utilize tags to enhance organization and clarity.

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+  ![descirption](btl-distance-des.png "Description of dataobject btl-disctance")
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
  ![description](btl-distance-des.png "Description of dataobject btl-distance")

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+  ![metadata-label](config-int_airports.png "Config view of int-airports with ID in Green and Labels from meatadata in brown ")
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
  ![metadata-label](config-int_airports.png "Config view of int-airports with ID in Green and Labels from metadata in brown ")

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 8]
```diff
@@ -0,0 +1,53 @@
...
+  
+
+* ### Clear Visualization of Transformations:
+  For actions involving transformations, our tool provides clear visualizations within the configuration. Easily comprehend input and output data for each action, with seamless links to corresponding data objects. This feature simplifies the understanding of complex data transformations, ensuring transparency and efficiency in your workflow.
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
  For actions involving transformations, our tool provides clear visualizations within the configuration. Easily comprehend input and output data for each action, with seamless links to corresponding data objects. This feature simplifies the understanding of complex data transformations, ensuring transparency about the business logic applied.

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+![transformer](transformer.png "Action View of historize-airports")
+
+* ### Data Lineage Graph:
+  Track and understand the flow of data objects across various processing stages with our built-in lineage feature. Visualize the relationships between data objects and actions involved in your data pipeline, empowering you to monitor the data's journey and make informed decisions.
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
  Track and understand the flow of data across various processing stages with our built-in lineage feature. Visualize the dependencies between data objects and actions involved in your data pipeline, empowering you to monitor the data's journey and answer questions about the impact of changes.

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 10]
```diff
@@ -0,0 +1,53 @@
...
+  ![lineage](config-int_airports_lineage.png "Lineage of dataobject int-airports with invloved action id")
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
  ![lineage](config-int_airports_lineage.png "Lineage of dataobject int-airports with involved action id")

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 11]
```diff
@@ -0,0 +1,53 @@
...
+
+Don't miss out on this game-changing tool! Embrace the future of data pipeline visualization and empower your team with the insights they need to excel. Get started today and explore the tool now on our [GitHub repository](https://github.com/smart-data-lake/sdl-visualization) and revolutionize the way you manage data pipelines.
```
Comment (zzeekk) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
Don't miss out on this game-changing tool for developing data pipelines with SDLB efficiently! Embrace our future of data pipeline visualization and empower your team with this simple but powerful tool. Get started today and following the installation guide on our [GitHub repository](https://github.com/smart-data-lake/sdl-visualization) and revolutionize the way you manage data pipelines.

```

File: docs/features.md [hunk 12]
```diff
@@ -103,3 +103,10 @@ More details on the roadmap will follow shortly.
 
 ## Housekeeping
 * Delete, or archive & compact partitions according to configurable expressions
+
+## UI Tool 
+* Comprehensive data flow dashboard
+* Config viewer with data catalog and DAG
```
Comment (zzeekk) [inline docs/features.md]:
```suggestion
* Config viewer with data catalog and dependency graph
```

File: docs/getting-started/config-examples/application-part3-with-meatdata.conf
Comment (zzeekk) [inline docs/getting-started/config-examples/application-part3-with-meatdata.conf:1]: Typo on Filename: meatdata -> metadata

File: docs/getting-started/part-3/metadata.md [hunk 13]
```diff
@@ -0,0 +1,82 @@
...
+id: metadata
+title: Metadata
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Goal
+
+In this part, we are going to dive into the role of metadata in the Smart Data Lake Builder (SDLB) and demonstrates how to define metadata for SDLB objects using the configuration file [solution](../config-examples/application-part3-download-incremental-mode.conf) presented in part 3. Metadata, in the context of Smart Data Lake, refers to additional information that goes beyond the basic name and description of DataObjects and Actions. It includes details such as feeds for Actions, tags associated with the component, and other attributes that help manage and explore data effectively.
```
Comment (zzeekk) [inline docs/getting-started/part-3/metadata.md]:
```suggestion
In this part, we are going to dive into the role of descriptive metadata in Smart Data Lake Builder (SDLB) and demonstrate how to define this additional metadata for SDLB objects using the configuration file [solution](../config-examples/application-part3-download-incremental-mode.conf) presented in part 3.
Descriptive metadata, in the context of SDLB, refers to additional configuration attributes and markdown files that goes beyond the basic "id" of DataObjects and Actions. It is mainly about adding descriptions and tags to configuration object, and other attributes that help manage and explore data effectively. 
```

File: docs/getting-started/part-3/metadata.md [hunk 14]
```diff
@@ -0,0 +1,82 @@
...
+          tags = ["aviation", "flight", "departures"]
+    }
+  }
+```
+:::tip UI Tool
+If we check [UI Demo](https://ui-demo.smartdatalake.ch/#/config/dataObjects/ext-departures), the given information in metadata are visualized for `ext-departures`.
+
+![ext-deparures-config](ext-deparures-config.png)
+
+For a deeper understanding of our visualization tool, please refer to our [bolg](../../../blog/sdl-uidemo).
```
Comment (zzeekk) [inline docs/getting-started/part-3/metadata.md]:
```suggestion
For a deeper understanding of our visualization tool, please refer to this [blog entry](../../../blog/sdl-uidemo).
```

File: docs/getting-started/part-3/metadata.md [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Goal
+
+In this part, we are going to dive into the role of metadata in the Smart Data Lake Builder (SDLB) and demonstrates how to define metadata for SDLB objects using the configuration file [solution](../config-examples/application-part3-download-incremental-mode.conf) presented in part 3. Metadata, in the context of Smart Data Lake, refers to additional information that goes beyond the basic name and description of DataObjects and Actions. It includes details such as feeds for Actions, tags associated with the component, and other attributes that help manage and explore data effectively. 
+
+While metadata attributes are optional, they can be very useful in many situations, as it is the case of *feeds* for actions.
```
Comment (zzeekk) [inline docs/getting-started/part-3/metadata.md]:
```suggestion
While metadata configuration attributes are optional, they can be very useful in many situations.
```

File: docs/getting-started/part-3/metadata.md [hunk 16]
```diff
@@ -0,0 +1,82 @@
...
+
+## Metadata in DataObject
+SDLB has file-based metadata feature, allowing users to provide additional information when defining DataObjects in the configuration file. This serves as more than just documentation, offering a comprehensive understanding and management of DataObjects by providing detailed descriptions, as well as labels for layers, subject areas, and tags
```
Comment (zzeekk) [inline docs/getting-started/part-3/metadata.md]: Mh, file-based metadata feature is misleading to me. Also this section is more like general explanation. I would just write about layers, subject areas and tags.

File: docs/getting-started/part-3/metadata.md [hunk 17]
```diff
@@ -0,0 +1,82 @@
...
+metadata {
+          name = "Airport Departures Join"
+          description = "merging flight details and airport locations"
+          tags = ["merge", "airports", "coordinates"]
+          feed = compute
+    }
+```
+
+* ### Metadata and Feeds
+SDLB acknowledges the need to run specific parts of defined pipelines. This is achieved by associating actions with *feeds*, which act as markers for sub-pipelines within a greater pipeline. Feeds inserted into metadata allow the selective execution of actions during development, debugging, or various use cases.
```
Comment (zzeekk) [inline docs/getting-started/part-3/metadata.md]: Feeds are also described here https://smartdatalake.ch/docs/getting-started/part-1/select-columns/#more-on-feeds
@pgruetter: Do you think we should we describe it again here?

File: docs/getting-started/part-3/metadata.md [hunk 18]
```diff
@@ -0,0 +1,82 @@
...
+  }
+}
+```
+
+Feeds in metadata are instrumental in selecting the right actions to run. During SDL execution, actions corresponding to certain feeds can be specified. Multiple actions can be associated with the same feed, allowing for more granular control over the execution of the data pipeline.
+
+## Try it out
+Now let us experiment with editing metadata for DataObjects and Actions from the [solution](../config-examples/application-part3-download-incremental-mode.conf) in the previous section. Customize metadata based on your understanding of different data, and modify feeds to run parts of the data pipeline selectively. The final configuration file could resemble [this](../config-examples/application-part3-with-meatdata.conf).
+
+In conclusion, metadata is the unsung hero of Smart Data Lake management, enabling organizations to not only organize and process data efficiently but also to ensure data quality and support advanced analytics. With its versatile applications, metadata proves to be a important player in the evolving landscape of data management.
```
Comment (zzeekk) [inline docs/getting-started/part-3/metadata.md]:
```suggestion
In conclusion, descriptive metadata is the unsung hero of data pipeline management, enabling teams to organize and document data more efficiently. 
```
In my opinion the last sentence it so much :-)

File: docs/getting-started/part-3/metadata.md
Comment (zzeekk) [inline docs/getting-started/part-3/metadata.md:1]: Should we also write about the markdown description files?

There is also a metadata.name which is not mentioned, where one can define a user friendly name for the configuration object.

And also Transformations can have a name and a description. In my opinion this is very useful to describe complex Actions with multiple transformations chained in more detail.

On the other side i would reduce the section on feeds, and just refer to the getting started chapter https://smartdatalake.ch/docs/getting-started/part-1/select-columns/#more-on-feeds

**Review by zzeekk — COMMENTED** [review]
I like the new fresh style!
Made some comments with improvements, and some points to discuss and "refactor".

Maybe you can first check and accept the improvement suggestions, and then we continue to discuss the open points with the others?

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md (hunk 2 above)
Comment (dust629) [reply blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]: I agree, the last sentence will be deleted.

File: docs/getting-started/part-3/metadata.md (hunk 16 above)
Comment (dust629) [reply docs/getting-started/part-3/metadata.md]: Thanks, this will be changed

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 19]
```diff
@@ -0,0 +1,62 @@
+---
+title: Exciting News! Introducing Our New UI Tool for Data Catalog Visualization
+description: Data Pipeline Visualization Tool
+slug: sdl-uidemo
+authors:
+  - name: Shasha Jiang
+    url: https://github.com/dust629
+tags: [DataCatalog, DataVisualization, DataFlow, DataLineage, DataManagement, DataPipeline, TechRevolution, GitHub OpenSource]
```
Comment (pgruetter) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
tags: [DataCatalog, DataVisualization, DataFlow, DataLineage, DataManagement, DataPipeline, GitHub OpenSource]

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 20]
```diff
@@ -0,0 +1,62 @@
...
+description: Data Pipeline Visualization Tool
+slug: sdl-uidemo
+authors:
+  - name: Shasha Jiang
+    url: https://github.com/dust629
+tags: [DataCatalog, DataVisualization, DataFlow, DataLineage, DataManagement, DataPipeline, TechRevolution, GitHub OpenSource]
+hide_table_of_contents: false
+---
+
+We are thrilled to announce the launch of our new [UI tool](https://ui-demo.smartdatalake.ch/) designed to revolutionize the way we visualize, track, and manage data pipelines! This addition to SDLB acts as a user-friendly technical data catalog that showcases dependencies through a graph, along with metadata about data pipeline execution (workflows). Our UI tool is a step forward in data pipeline management, providing enhanced visibility and control. Whether you are a data engineer, analyst, or scientist, this tool will enhance your abilities to management data pipelines, allowing you to be more productive and achieve exceptional results.
```
Comment (pgruetter) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
We are thrilled to announce the launch of our new [UI tool](https://ui-demo.smartdatalake.ch/) designed to revolutionize the way we visualize, track, and manage data pipelines! This addition to SDLB acts as a user-friendly technical data catalog that showcases dependencies through a graph, along with metadata about data pipeline execution (workflows). Our UI tool is a step forward in data pipeline management, providing enhanced visibility and control. Whether you are a data engineer, analyst, or scientist, this tool will enhance your abilities to manage data pipelines, allowing you to be more productive and achieve exceptional results.

```

File: blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md [hunk 21]
```diff
@@ -0,0 +1,62 @@
...
+Get a bird's eye view of your entire workflow! Our intuitive dashboard visualizes intricate details for each run, including execution time, duration, and the feed configuration used. Deep dive into the timeline of involved actions, with the ability to access detailed data by simply clicking on each action.
+![Timeline of Action](timeline.png "Timeline of Action: Description or Caption")
+
+
+## 🔍 Config Viewer with Rich Content
+Explore an expansive Configuration Viewer that contains all the information from your config file, and extra information you provide as documentation. You can find:
+
+
+* ###  Markdown Descriptions:
+  SDLB supports creating descriptions as markdown files, enabling you to provide detailed information about actions or objects. These descriptions can include images, hyperlinks, and other formatting options. They serve as excellent documentation that is versionned together with your code, aiding your understanding of the workflow's structure and function. Moreover, you can utilize tags to enhance organization and clarity.
```
Comment (pgruetter) [inline blog/2023-11-09-UI-Demo/2023-11-09-UI-Demo.md]:
```suggestion
  SDLB supports creating descriptions as markdown files, enabling you to provide detailed information about actions or objects. These descriptions can include images, hyperlinks, and other formatting options. They serve as excellent documentation that is versioned together with your code, aiding your understanding of the workflow's structure and function. Moreover, you can utilize tags to enhance organization and clarity.

```

File: docs/getting-started/part-3/metadata.md
Comment (pgruetter) [reply docs/getting-started/part-3/metadata.md:1]: :+1:

File: docs/getting-started/part-3/metadata.md [hunk 22]
```diff
@@ -0,0 +1,87 @@
...
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+## Goal
+
+In this part, we are going to dive into the role of descriptive metadata in Smart Data Lake Builder (SDLB) and demonstrate how to define this additional metadata for SDLB objects using the configuration file [solution](../config-examples/application-part3-download-incremental-mode.conf) presented in part 3.
+
+## About Metadata 
+<!--Should we also write about the markdown description files?-->
+Descriptive metadata, in the context of SDLB, refers to additional configuration attributes and markdown files that goes beyond the basic "id" of DataObjects and Actions. It is mainly about adding descriptions and tags to configuration object, and other attributes that help manage and explore data effectively.
```
Comment (pgruetter) [inline docs/getting-started/part-3/metadata.md]:
```suggestion
Descriptive metadata, in the context of SDLB, refers to additional configuration attributes and markdown files that go beyond the basic "id" of DataObjects and Actions. It is mainly about adding descriptions and tags to configuration object, and other attributes that help manage and explore data effectively. 
```

File: docs/getting-started/part-3/metadata.md [hunk 23]
```diff
@@ -0,0 +1,87 @@
...
+In this part, we are going to dive into the role of descriptive metadata in Smart Data Lake Builder (SDLB) and demonstrate how to define this additional metadata for SDLB objects using the configuration file [solution](../config-examples/application-part3-download-incremental-mode.conf) presented in part 3.
+
+## About Metadata 
+<!--Should we also write about the markdown description files?-->
+Descriptive metadata, in the context of SDLB, refers to additional configuration attributes and markdown files that goes beyond the basic "id" of DataObjects and Actions. It is mainly about adding descriptions and tags to configuration object, and other attributes that help manage and explore data effectively. 
+
+## Metadata and Descriptions
+<!--There is also a metadata.name, where one can define a user friendly name for the configuration object. -->
+
+While metadata configuration attributes are optional, they can be very useful in many situations. `metadata.name` can define a user frinedly and name stringforward for the configuration object.
```
Comment (pgruetter) [inline docs/getting-started/part-3/metadata.md]:
```suggestion
While metadata configuration attributes are optional, they can be very useful in many situations. `metadata.name` can define a user friendly and name stringforward for the configuration object. 
```

Comment (pgruetter) [inline docs/getting-started/part-3/metadata.md]: Not sure what is meant with "can define a user friendly and name stringforward for the configuration object"

Do you mean "can define a user friendly and straight forward name for the configuration object" ?

File: docs/getting-started/part-3/metadata.md [hunk 24]
```diff
@@ -0,0 +1,87 @@
...
+:::tip UI Tool
+If we check [UI Demo](https://ui-demo.smartdatalake.ch/#/config/dataObjects/ext-departures), the given information in metadata are visualized for `ext-departures`.
+
+![ext-deparures-config](ext-deparures-config.png)
+
+For a deeper understanding of our visualization tool, please refer to this [blog entry](../../../blog/sdl-uidemo).
+:::
+
+## Metadata and Feeds
+<!--On the other side i would reduce the section on feeds, and just refer to the getting started chapter https://smartdatalake.ch/docs/getting-started/part-1/select-columns/#more-on-feeds -->
```
Comment (pgruetter) [inline docs/getting-started/part-3/metadata.md]: Agree, I would remove this section and just mention that the feed we already had configured, is displayed in the UI too.

**Review by pgruetter — APPROVED** [review]

**Review by pgruetter — APPROVED** [review]
Added a few more suggestions.

File: docs/getting-started/part-3/metadata.md (hunk 23 above)
Comment (dust629) [reply docs/getting-started/part-3/metadata.md]: yes i will change it

**Pull request merged**